src/db.c

```diff
@@ -1693,14 +1695,21 @@ void setExpire(client *c, serverDb *db, robj *key, long long when) {
 
 /* Return the expire time of the specified key, or -1 if no expire
  * is associated with this key (i.e. the key is non volatile) */
-long long getExpire(serverDb *db, robj *key) {
+long long getExpireWithSlot(serverDb *db, robj *key, int slot) {
     dictEntry *de;
 
-    if ((de = dbFindExpires(db, key->ptr)) == NULL) return -1;
+    if ((de = dbFindExpiresWithSlot(db, key->ptr, slot)) == NULL) return -1;
 
     return dictGetSignedIntegerVal(de);
 }
 
+/* Return the expire time of the specified key, or -1 if no expire
+ * is associated with this key (i.e. the key is non volatile) */
+long long getExpire(serverDb *db, robj *key) {
+    int slot = server.cluster_enabled ? getKeySlot(key->ptr) : 0;
```

Comment:
```suggestion
    int dict_index = getKVStoreIndexForKey(key->ptr);
```
Throughout the codebase I think the preference was to call this KVStore index as opposed to slot. It is the slot for cluster mode, but it's not the slot in CMD.

Reply:
I'll update the usage